### PR TITLE
[TASK] Bump the production dependencies

### PR DESCRIPTION
**Pull request by oliverklee**
The required PHPStan version is lower than the currently latest version
in order to allow installations together with the latest release of
`ssch/typo3-rector` (which has a hard dependency on `rector/rector:0.17.0`,
which in turn requires `phpstan/phpstan:^1.10.15`).

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -12,12 +12,12 @@
...
-		"phpstan/phpstan": "^1.8.9",
-		"nikic/php-parser": "^4.15.1",
+		"phpstan/phpstan": "^1.10.27",
+		"nikic/php-parser": "^4.18.0",
 		"typo3/cms-core": "^10.4 || ^11.5 || ^12.4",
 		"typo3/cms-extbase": "^10.4 || ^11.5 || ^12.4",
-		"bnf/phpstan-psr-container": "^1.0",
-		"composer/semver": "^3.3"
+		"bnf/phpstan-psr-container": "^1.0.1",
+		"composer/semver": "^3.4.0"
```
Comment (sascha-egerer) [inline composer.json:20]: You're now only increasing the minimum version. I don't see why this should be required. 
`^1.8.9` means `>=1.8.9 && < 2.0.0`. So i don't see why this change should be required for all the dependencies. Are am i wrong?

**Review by sascha-egerer — CHANGES_REQUESTED** [review]

File: composer.json (hunk 1 above)
Comment (oliverklee) [reply composer.json:20]: Raising the minimum version of a dependency allows us to rely on the bugfixes and features provided by the latest version. (If we don't need those, we can keep the dependencies as-is.)

Comment (sascha-egerer) [reply composer.json]: > Raising the minimum version of a dependency allows us to rely on the bugfixes and features provided by the latest version. 
Sure, but at the same time we prevent users from updating if their dependencies are not up to date (due to reasons?). So i would like to prevent as much as possible to defined "new" versions as fixed requirements if not necessary.

Comment (oliverklee) [reply composer.json]: Okay, closing then.

Comment (sascha-egerer) [reply composer.json]: Ok thanks. Maybe we can plan a bigger release once TYPO3 13 LTS is out and drop most of the old stuff as a lot of stuff is already solved in Core now anyways. But lets see how it goes on
